### ISSUE TITLE
UD parse has "compound" instead of "name"

### ISSUE THREAD

**Issue by nschneid** [stanfordnlp/CoreNLP#219]
@ida-szubert has discovered that CoreNLP, including the [online demo](http://corenlp.run/), systematically produces `compound` dependencies which should be [`name`](http://universaldependencies.org/en/dep/name.html) relations.

E.g.: For "I met Carl XVI Gustaf.", even though the NER correctly recognizes "Carl XVI Gustaf" as a PERSON, the basic and enhanced++ UD parses have `compound(Gustaf, Carl)` and `compound(Gustaf, XVI)`.

Is this fixable?

**Comment by nschneid** [comment]
@sebschu

**Comment by sebschu** [comment]
Yes, we are currently not using NER tags in the converter and therefore we also cannot distinguish between `compound` and `name` relations. It's on my todo-list, and I'll try to get to this within the next couple of days.

**Comment by nschneid** [comment]
Excellent, a fix would be much appreciated.

Note that the treatment of some aspects of names is under discussion in UniversalDependencies/docs#324 and UniversalDependencies/docs#253.

**Comment by sebschu** [comment]
So, I implemented a first version of this in 10f4d06e6bbf3410001ee854b88221e84a39c4e3. However, as I haven't updated any of the downstream annotators yet, I made this optional, and by default, we still always output the `compound` relation for flat NPs.

If you want to use the `name` relation in CoreNLP or in `UniversalDependenciesConverter`, then add the new `UDUseNameRelation` Java property to your command.

For example:

    java -DUDUseNameRelation edu.stanford.nlp.trees.ud.UniversalDependenciesConverter -treeFile trees.txt

(Note that this property has to go before the class name.)

We use NER tags to distinguish between `name` and `compound` relations, so make sure that you run the `ner` annotator before the `parse` annotator if you are running CoreNLP. And if you run the converter, make sure that the models .jar is in the classpath. (The converter will automatically add NER tags to each sentence).

The code currently only outputs the `name` relation for person and location names, but not for names of organizations because in many cases `compound` is the correct relation. (e.g., _Natural Resources Conservation Service_). Let me know if you have any ideas on how to get these right as well. (My best idea so far: Training a sequence model that is able to distinguish proper organization names and organization names that are actual compound nouns, but I'm not sure how feasible that is).

Hope this is useful for now and let me know if you run into any issues/bugs.
